sed/loader/flash/buffer_handler.py

```diff
@@ -79,160 +129,137 @@ def _schema_check(self) -> None:
-            df_electron, meta = split_dld_time_from_sector_id(
-                df_electron,
-                config=self._config,
+            # Forward fill the non-electron channels across files
+            overlap = min(file["num_rows"] for file in file_stats[typ].values())
+            iterations = self._config.get("forward_fill_iterations", 2)
+            df = forward_fill_lazy(
+                df=df,
+                columns=self.fill_channels,
+                before=overlap,
+                iterations=iterations,
             )
-            self.metadata.update(meta)
+            # TODO: This dict should be returned by forward_fill_lazy
+            filling[typ] = {
+                "columns": self.fill_channels,
+                "overlap": overlap,
+                "iterations": iterations,
+            }
 
-        self.df_electron = df_electron.astype(dtypes)
-        self.df_pulse = dataframe[index + fill_channels]
+            self.df[typ] = df
+        self.metadata.update({"file_statistics": file_stats, "filling": filling})
 
     def run(
         self,
         h5_paths: list[Path],
         folder: Path,
         force_recreate: bool = False,
-        prefix: str = "",
         suffix: str = "",
         debug: bool = False,
     ) -> None:
         """
         Runs the buffer file creation process.
+        Does a schema check on the buffer files and creates them if they are missing.
+        Performs forward filling and splits the sector ID from the DLD time lazily.
 
         Args:
             h5_paths (List[Path]): List of paths to H5 files.
-            folder (Path): Path to the folder for buffer files.
+            folder (Path): Path to the folder for processed files.
             force_recreate (bool): Flag to force recreation of buffer files.
-            prefix (str): Prefix for buffer file names.
             suffix (str): Suffix for buffer file names.
             debug (bool): Flag to enable debug mode.):
         """
-
-        self._get_files_to_read(h5_paths, folder, prefix, suffix, force_recreate)
+        self.fp = BufferFilePaths(h5_paths, folder, suffix)
 
         if not force_recreate:
-            self._schema_check()
-
-        self._save_buffer_files(debug)
+            schema_set = set(
+                get_channels(self._config["channels"], formats="all", index=True, extend_aux=True),
+            )
+            self._schema_check(self.fp["electron"], schema_set)
+            schema_set = set(
+                get_channels(
+                    self._config["channels"],
+                    formats=["per_pulse", "per_train"],
+                    index=True,
+                    extend_aux=True,
+                ),
+            ) - {"electronId"}
+            self._schema_check(self.fp["timed"], schema_set)
+
+        self._save_buffer_files(force_recreate, debug)
 
         self._fill_dataframes()
+
+        # Correct the 3-bit shift which encodes the detector ID in the 8s time
```

Comment:
your naming is somewhat misleading here. You save the buffer files, but you never load them...

Reply:
Right. Can be fixed.